### ISSUE TITLE
why shall `num_inputs` match `num_val_inputs` ?

### ISSUE THREAD

**Issue by WladimirSidorenko** [ncullen93/torchsample#95]
What's the intuition behind this assert? Validation set is usually much smaller than training data.

https://github.com/ncullen93/torchsample/blob/21507feb258a25bf6924e4844e578624cda72140/torchsample/modules/module_trainer.py#L321

**Comment by luyuhua** [comment]
+1
why?

**Comment by luyuhua** [comment]
it should change like this

        if isinstance(loader.dataset,th.utils.data.dataset.Dataset):
            num_inputs = num_targets = 1
        else:
            num_inputs = loader.dataset.num_inputs
            num_targets = loader.dataset.num_targets
